pkg/openstack/common.go

```diff
@@ -7,12 +7,32 @@ import (
 
 	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
 	"github.com/openstack-k8s-operators/lib-common/modules/common/util"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
 	apimeta "k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// CheckDeleteDisabled - Delete the object which in turn will clean the sub resources
+func CheckDeleteDisabled(ctx context.Context, helper *helper.Helper, obj client.Object) (ctrl.Result, error) {
+	key := client.ObjectKeyFromObject(obj)
+	if err := helper.GetClient().Get(ctx, key, obj); err != nil {
+		if k8s_errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+	// Delete the object
+	if obj.GetDeletionTimestamp().IsZero() {
+		if err := helper.GetClient().Delete(ctx, obj); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+	return ctrl.Result{}, nil
+
+}
+
```

Comment:
There's a lot of talk about enabling/disabling services going on in another active PR [1] for this operator.  Perhaps at this point we need to reconcile what is happening there with these "disable-and-delete" logic changes here. @bshephar @gibizer thoughts?

[1] https://github.com/openstack-k8s-operators/openstack-operator/pull/162

Reply:
I guess that one is meant to update the 'status', as disabling services leads to the status of openstackcontrolplane instance not updated to "Setup complete" and won't change with this.

Reply:
> need to reconcile

I start to feel like an k8s operator :)

Thanks @abays for spotting the parallel tracks. 

I think we can adjust this patch to to solve the problem in the other PR as well. See above by comment about removing the service specific condition from the condition list when the service is deleted.

Reply:
It seems like there is some common ground between these two PRs, specifically with regards to one of the latest comments from @gibizer:

```
I can see one solution that solves all problems:

    Solution1: When a service is disabled then delete the service CR and after that succeeded, delete the service specific condition from OpenStackControlPlane.Status.Condition.
```

The whole comment [1] can provide more context.  It seems like some people might have different opinions as to what should happen to a service when it is disabled.

[1] https://github.com/openstack-k8s-operators/openstack-operator/pull/162#issuecomment-1408224224

Reply:
Yeah I can see what you mean now.